tests/test_backup.py

```diff
@@ -110,3 +112,228 @@ def test_backup_vault_kms_filter(self):
+        backup_vault_arn = backup_vault["BackupVaultArn"]
+
+        def cleanup():
+            client.delete_backup_vault(BackupVaultName=backup_vault_name)
+            if self.recording:
+                time.sleep(60)
+
+        self.addCleanup(cleanup)
+
+        client.put_backup_vault_access_policy(
+            BackupVaultName=backup_vault_name,
+            Policy=json.dumps(
+                {
+                    'Version': '2012-10-17',
+                    'Statement': [
+                        {
+                            'Sid': 'SpecificAllow',
+                            'Effect': 'Allow',
+                            'Principal': '*',
+                            'Action': ['backup:GetBackupVaultAccessPolicy'],
+                            "Resource": backup_vault_arn,
+                        },
+                        {
+                            "Sid": "RemoveMe",
+                            "Effect": "Allow",
+                            "Principal": {"AWS": "arn:aws:iam::123456789123:root"},
+                            "Action": ["backup:PutBackupVaultAccessPolicy"],
+                            "Resource": backup_vault_arn,
+                        }
+                    ],
+                }
+            )
+        )
+
+        p = self.load_policy(
+            {
+                "name": "test-backup-vault-modify-policy-add-statements",
+                "resource": "backup-vault",
+                "filters": [
+                    {"BackupVaultName": backup_vault_name}
+                ],
+                "actions": [
+                    {
+                        "type": "modify-policy",
+                        "add-statements": [],
+                        "remove-statements": ["RemoveMe"],
+                    }
+                ],
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        if self.recording:
+            time.sleep(30)
+
+        self.assertEqual(len(resources), 1)
+        data = json.loads(
+            client.get_backup_vault_access_policy(
+                BackupVaultName=resources[0]["BackupVaultName"]
+            )["Policy"]
+        )
+
+        self.assertTrue("SpecificAllow" in [s["Sid"] for s in data.get("Statement", ())])
+        self.assertFalse("RemoveMe" in [s["Sid"] for s in data.get("Statement", ())])
+
+    def test_backup_vault_modify_add_remove_statements(self):
+        session_factory = self.replay_flight_data(
+            "test_backup_vault_modify_add_remove_statements"
+        )
+        client = session_factory().client("backup")
+        backup_vault = client.create_backup_vault(
+            BackupVaultName="test_backup_vault_modify_add_remove_statements"
+        )
+
+        backup_vault_name = backup_vault["BackupVaultName"]
+        backup_vault_arn = backup_vault["BackupVaultArn"]
+
+        def cleanup():
+            client.delete_backup_vault(BackupVaultName=backup_vault_name)
```

Comment:
we're not doing this anymore, re creating resources in tests, it adds extraneous data recordings without adding value.